src/Control/Effect/Parse.hs

```diff
@@ -37,14 +39,23 @@ parse :: (Member Parse sig, Carrier sig m)
 parse parser blob = send (Parse parser blob pure)
 
 
+-- | Select a parser for the given 'Language'.
+parserForLanguage :: Map.Map Language (SomeParser c ann) -> Language -> Maybe (SomeParser c ann)
+parserForLanguage = flip Map.lookup
+
+-- | Select a parser for the given 'Blob'.
+parserForBlob :: Map.Map Language (SomeParser c ann) -> Blob -> Maybe (SomeParser c ann)
+parserForBlob parsers = parserForLanguage parsers . blobLanguage
```

Comment:
These help us avoid having to import `Data.Map` in too many places, and could also be used to completely abstract parser maps (tho I don’t think we particularly need to).